### PR TITLE
Vpngw generation add

### PR DESCRIPTION
**Pull request by AdamBrassard**
##### SUMMARY

I needed the ability to set the Generation of the VirtualNetworkGateway and wanted to contribute this to the community.
Adding the ability to set Virtual Network Gateway Generation in azure_rm_virtualnetworkgateway module. 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

plugins/module_utils/azure_rm_common.py
plugins/modules/azure_rm_virtualnetworkgateway.py
tests/integration/targets/azure_rm_virtualnetworkgateway/tasks/main.yml

##### ADDITIONAL INFORMATION

- Adding support for setting VPNGW generation
- Updated azure_rm_common.py NetworkManagementClient version to 2019-11-01
- Updated azure_rm_virtualnetworkgateway.py for additional input and choices of VPNGW Generation options as outlined in SDK
- Updated integration test to generate public ip name in same method as vngname for multiple runs
- Updated integration tests adding block for gen setting and clean up of VNET/SUBNET/PUBIP after test and delete of VPNGW

```yaml
- name: Create virtual network gateway without bgp settings
  azure_rm_virtualnetworkgateway:
    resource_group: "{{ resource_group }}"
    name: "{{ vngname }}"
    ip_configurations:
      - name: testipconfig
        private_ip_allocation_method: Dynamic
        public_ip_address_name: "{{ pubipname }}"
    virtual_network: "{{ vnetname }}"
    tags:
      common: "xyz"
```
```yaml
- name: Create virtual network gateway Generation2 (check mode)
  check_mode: yes
  azure_rm_virtualnetworkgateway:
    resource_group: "{{ resource_group }}"
    name: "{{ vngname }}"
    sku: VpnGw2
    vpn_gateway_generation: Generation2
    ip_configurations:
      - name: testipconfig
        private_ip_allocation_method: Dynamic
        public_ip_address_name: "{{ pubipname }}" #testPublicIP
    virtual_network: "{{ vnetname }}"
    tags:
      common: "xyz"
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Change Log
 
+## v1.13.1 (2022-07-14)
```
Comment (Fred-sun) [inline CHANGELOG.md]: Please delete 3~7 lines. It is not a major change and will not be released for a separate feature. So please cancel, we will regularly post all recent changes together. Thank you very much!

Comment (AdamBrassard) [reply CHANGELOG.md]: Of course, thank you. This has been removed

File: galaxy.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ namespace: azure
 name: azcollection
 
 # The version of the collection. Must be compatible with semantic versioning
-version: 1.13.0
+version: 1.13.1
```
Comment (Fred-sun) [inline galaxy.yml]: Now we release a version regularly, the next version should be 1.14.0. Thank you!
```suggestion
version: 1.13.0
```

**Comment by xuzhang3** [comment]
@AdamBrassard LGTM

**Pull request merged**